project/PekkoBuild.scala

```diff
@@ -133,6 +133,7 @@ object PekkoBuild {
   final val DefaultJavacOptions = Seq("-encoding", "UTF-8", "-Xlint:unchecked", "-XDignore.symbol.file")
 
   lazy val defaultSettings: Seq[Setting[_]] = Def.settings(
+    projectInfoVersion := (if (isSnapshot.value) "snapshot" else version.value),
```

Comment:
This trick was taken from pekko-grpc (see https://github.com/apache/incubator-pekko-grpc/blob/f43a3a02998f26ca16b6d5e311f1378ad330810f/project/Common.scala#L46). This allows the `link-validator` to work on PR's because it sets the version to `snaphot` (if the current version of sbt is a snapshot) allowing links to scala/java docs to resolve, i.e. `https://pekko.apache.org/api/pekko/snapshot/org/apache/pekko/snapshot/ClusterLogMarker$.html` (notice the `snapshot` being used in the version)